### PR TITLE
Labels for Registers

### LINKED ISSUE #347: Simulations Index Register Negative Numbers
When storing a number `x`, it should store `x % this.size` to prevent over and underflow.

### PR DESCRIPTION
**Pull request by equationcrunchor**
Adds support for labeling registers.  

Resolves #347, and #239.

### CODE REVIEW

File: static/js/render.js [hunk 1]
```diff
@@ -207,11 +207,77 @@ export class DemoRenderer extends Renderer {
...
+        numbers_area.classList.add("form-control");
+        numbers_area.classList.add("code_box");
+        numbers_area.classList.add("code_numbers");
+        numbers_area.setAttribute("rows", "1");
+        numbers_area.setAttribute("id", "code_numbers_" + this.num_lines.toString());
+        numbers.appendChild(numbers_area);
+        line.appendChild(numbers);
+
+        code.appendChild(line);
+        this.num_lines++;
```
[the remark this answers is not in the document]
Comment (equationcrunchor) [reply static/js/render.js]: Is this a concern for this project?

[the remark this answers is not in the document]
Comment (mscuthbert) [reply static/js/render.js]: No, but just as long as I was making review comments, I thought you should be aware -- most of the eslint configs I've seen from corps (esp. AirBnB) forbid.

**Comment by equationcrunchor** [comment]
@mscuthbert Made changes; should be ready for review again!

**Comment by equationcrunchor** [comment]
@mscuthbert Selenium tests pass now

**Pull request merged**